### PR TITLE
3002.1

### PR DESCRIPTION
**Pull request by dwoz**
Merge forward CVE fixes.

### CODE REVIEW

File: changelog/cve-2020-16846.fixed [hunk 1]
```diff
@@ -0,0 +1 @@
+CVE-2020-16804 - Prevent shell injections in netapi ssh client
```
Comment (meaksh) [inline changelog/cve-2020-16846.fixed]: Just noticed a typo on the CVE reference: CVE-2020-168**4**6

File: salt/client/ssh/shell.py [hunk 2]
```diff
@@ -31,22 +32,15 @@
...
-    pass
-
-
 def gen_key(path):
     """
     Generate a key for use with salt-ssh
     """
-    cmd = 'ssh-keygen -P "" -f {0} -t rsa -q'.format(path)
-    if not os.path.isdir(os.path.dirname(path)):
+    cmd = ["ssh-keygen", "-P", '""', "-f", path, "-t", "rsa", "-q"]
```
Comment (nadvornik) [inline salt/client/ssh/shell.py:39]: This changes the passphrase from empty string to a string containing quotes. Is it really intended?

Comment (agraul) [reply salt/client/ssh/shell.py:39]: Just to explain why this happens: `cmd` always contained `-P ""`, the difference is that it was passed to a shell which looked at `""` and saw it as an empty string. Now that the `subprocess.call` is without `shell=True`, nothing is in between this line and `ssh-keygen` that could strip the `""`.  
To fix this, it should be enough to remove the `'""'` list element.

Comment (dwoz) [reply salt/client/ssh/shell.py:39]: Agreed, I will update this.

File: changelog/cve-2020-16846.fixed (hunk 1 above)
Comment (dwoz) [reply changelog/cve-2020-16846.fixed]: Good catch, I'll fix this.

**Comment by bryceml** [comment]
is  tests/integration/files/conf/roaster supposed to be  tests/integration/files/conf/roster ?

**Comment by dwoz** [comment]
> is tests/integration/files/conf/roaster supposed to be tests/integration/files/conf/roster ?

Yeah, I misspelled it in the tests to.. Fixed now.

**Comment by dwoz** [comment]
re-run all

**Review by krionbsd — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**